File: src/codegen/errorMapping.ts
// JS error result mapping strategy
export function wrapOk(expr: string): string {
  return `Ok(${expr})`;
}

export function wrapErr(variant: string, expr?: string): string {
  return `Err("${variant}"${expr ? ", " + expr : ""})`;
}

export function unwrapOrPropagate(varName: string): string {
  return `__tangle_propagate(${varName})`;
}

File: compiler/tangle-cli/src/codegen/error_mapping.rs
pub fn wrap_ok(expr: &str) -> String {
    format!("Ok({})", expr)
}

pub fn wrap_err(variant: &str, expr: Option<&str>) -> String {
    match expr {
        Some(e) => format!("Err('{}', {})", variant, e),
        None => format!("Err('{}')", variant),
    }
}

pub fn unwrap_or_propagate(var_name: &str) -> String {
    format!("__tangle_propagate({})", var_name)
}
